feat: pokemon details

Miqqy0069 opened the pull request:
- added pokemon details connector, vm and page
- created reusable pokemon details cards
- added spacing constants
- ran analyze with 0 issues
- dart format with 0 changes

(analyze) 
![image](https://user-images.githubusercontent.com/75653480/228209815-5e2424e9-d5ac-4cf8-808f-65a18dbc5969.png)

(dart format) 
![image](https://user-images.githubusercontent.com/75653480/228209863-77583015-8bb4-4bcb-9cfc-04b19fd3d703.png)

(working app)

https://user-images.githubusercontent.com/75653480/228209964-034ad44a-c705-4508-bdfd-38c3e962a318.mp4

Miqqy0069 committed:
feat: pokemon details
- added pokemon details connector, vm and page
- created reusable pokemon details cards
- added spacing constants
- ran analyze with 0 issues
- dart format with 0 changes

shawn-ffufm commented on lib/main.dart:
```diff
@@ -13,7 +15,15 @@ void main() {
   );
 }
+
+final routes = {
```
kindly move this in a separate dart file 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/main.dart:
```diff
@@ -13,7 +15,15 @@ void main() {
 }
+
+final routes = {
+  '/': (BuildContext context) => const PokemonOverviewConnector(),
```
rename context to _ since it's unused 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/main.dart:
```diff
@@ -13,7 +15,15 @@ void main() {
+
+final routes = {
+  '/': (BuildContext context) => const PokemonOverviewConnector(),
+  PokemonDetailsConnector.routeName: (BuildContext context) => const PokemonDetailsConnector()
```
also here

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/state/action/pokemon_actions.dart:
```diff
@@ -32,3 +34,9 @@ class GetPokemonDetailsAction extends ReduxAction<AppState> {
   }
 }
+
+/// Dispose PokemonDetailsAction on store
```
kindly improve this documentation 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/spacing.dart:
```diff
@@ -0,0 +1,21 @@
+import 'package:flutter/material.dart';
+
+class HorizontalSpace extends StatelessWidget {
+  final double width;
```
move this below the constructor 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/spacing.dart:
```diff
@@ -0,0 +1,21 @@
+  final double width;
+  const HorizontalSpace({
+    super.key,
+    required this.width,
```
please make required properties always on top

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/spacing.dart:
```diff
@@ -0,0 +1,21 @@
+}
+
+class VerticalSpace extends StatelessWidget {
+  final double height;
```
also here

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/spacing.dart:
```diff
@@ -0,0 +1,21 @@
+    super.key,
+    required this.width,
+  });
+  @override
```
kindly add new line on top of this 🙂

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/spacing.dart:
```diff
@@ -0,0 +1,21 @@
+  final double height;
+  const VerticalSpace({
+    super.key,
+    required this.height,
```
same here

Miqqy0069 replied:
resolved

shawn-ffufm commented on lib/utilities/spacing.dart:
```diff
@@ -0,0 +1,21 @@
+    super.key,
+    required this.height,
+  });
+  @override
```
also here

Miqqy0069 replied:
resolved

shawn-ffufm reviewed (changes requested):
see pr feedback 🙂

Miqqy0069 committed:
feat: pokemon details
- pr feedback
- renamed unused contexts to _
- placed required properties on top of constructor
- adjusted spacing file

shawn-ffufm reviewed (approved):
lgtm
